giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/tokens/impl/IntrospectTokenService.java

```diff
@@ -0,0 +1,122 @@
+package edu.asu.diging.gilesecosystem.web.tokens.impl;
+
+import java.util.Date;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import edu.asu.diging.gilesecosystem.util.properties.IPropertiesManager;
+import edu.asu.diging.gilesecosystem.web.config.SystemMessageHandlerConfig;
+import edu.asu.diging.gilesecosystem.web.exceptions.ServerMisconfigurationException;
+import edu.asu.diging.gilesecosystem.web.service.properties.Properties;
+import edu.asu.diging.gilesecosystem.web.tokens.IApiTokenContents;
+import edu.asu.diging.gilesecosystem.web.tokens.IIntrospectTokenService;
+
+/**
+ * Class to introspect access token for MITREid connect server
+ * 
+ * @author snilapwa
+ *
+ */
+@Service
+public class IntrospectTokenService implements IIntrospectTokenService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private IPropertiesManager propertyManager;
+
+    @Autowired
+    private RestTemplate accessTokenRestTemplate;
+
+    @Autowired
+    private SystemMessageHandlerConfig systemMessageHandler;
+
+    /**
+     * Calls MITREid connect server introspect api using client with protected
+     * resource access keys.
+     * 
+     * @param accessToken accessToken to pass to introspect Api
+     * @return tokenContents with username and expiration status
+     * @throws ServerMisconfigurationException
+     */
+    public IApiTokenContents introspectAccessToken(String accessToken) throws ServerMisconfigurationException {
```

Comment:
this javadoc is not seen when someone autowires the dependency. Since they would be using the interface, only interface javadoc is seen.